clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventProcessor.java

```diff
@@ -320,11 +320,12 @@ private void process(final TopicPatternSubscriptionChangeEvent event) {
      * This will make the consumer send the updated subscription on the next poll.
      */
     private void process(final UpdatePatternSubscriptionEvent event) {
+        if (!subscriptions.hasPatternSubscription()) {
```

Comment:
Is this test really necessary? This event is only sent if the application thread knows there's a pattern subscription.

Reply:
Just checking my understanding. It looks like the remains of an earlier thought :)